docker/px4-dev/scripts/entrypoint.sh

```diff
@@ -4,6 +4,11 @@
 # Either use the LOCAL_USER_ID if passed in at runtime or
 # fallback
 
+if [ "${ROS_DISTRO}" == "melodic" ]; then
+	export DISPLAY=:0
```

Comment:
I'm confused. You're setting the display server to 0, but creating the virtual display at 1.